### PR TITLE
Add batch execution of tests

### PR DESCRIPTION
**Pull request by bryan-aguilar**
**Description:** 
**Please do not merge this PR**
This will be a multi PR change to add the ability to generate and execute a batch of test cases for the ADOT Collector. I would like to merge them all at once in case we have to cherry pick this into previous release branches. Having a single merge commit with all changes will reduce the chance that anything is missed. We could also merge this into a dev branch if it is easier that way.

This is Part 1 of N.

This PR adds a make target to execute the tests along with a shell script to parse the `test-case-batch` file 
and execute the `terraform apply`. 

A tool to automatically generate the `test-case-batch` file will be added in subsequent PRs.

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: docs/run-testing-framework.md [hunk 1]
```diff
@@ -213,6 +213,44 @@ Don't forget to clean up your resources:
 terraform destroy -auto-approve
 ````
 
+### 2.7 Run batch tests
+
+Batch testing allows a set of tests to be ran synchronously. To do this,
+a `test-case-batch` file is required in the `./terraform` directory. 
+The format of the `test-case-batch` file is as such. The `test-case-batch` 
+file can contain an infinite amount of test cases as
```
Comment (Aneurysm9) [inline docs/run-testing-framework.md]: Infinite?  Surely it's at least bounded by available disk space.

File: terraform/Makefile [hunk 2]
```diff
@@ -1,5 +1,22 @@
+ifeq ($(origin .RECIPEPREFIX), undefined)
+  $(error This Make does not support .RECIPEPREFIX. Please use GNU Make 4.0 or later)
```
Comment (Aneurysm9) [inline terraform/Makefile]: Why?  This seems unnecessary and impedes portability.  I wouldn't be able to run this locally as I have `GNU Make 3.81`.

Comment (bryan-aguilar) [reply terraform/Makefile]: The argument was found in this [post](https://tech.davis-hansson.com/p/make/) as I was doing some `make` research. The gist is that the recipe prefix can avoid any errors brought by tabs or spaces.

File: docs/run-testing-framework.md (hunk 1 above)
Comment (bryan-aguilar) [reply docs/run-testing-framework.md]: Updated

**Pull request closed without merging**